src/netaddress.h

```diff
@@ -107,6 +107,19 @@ class CNetAddr
         }
 
         friend class CSubNet;
+        friend struct CNetAddrHash;
+};
+
+struct CNetAddrHash
```

Comment:
You'll want to use a salted hash here (use CSipHasher from crypto/siphash.h or a specialization; see SaltedOutpointHasher in coins.h for inspiration). Otherwise an attacker may be able to degrade our performance by giving us a bunch of entries that all end up in the same bucket).

Reply:
Done, thanks for the suggestion!